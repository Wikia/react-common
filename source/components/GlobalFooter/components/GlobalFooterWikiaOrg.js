import React from 'react';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';

import Svg from './Svg';

const GlobalFooterWikiaOrg = ({ t, model }) => {
    return <React.Fragment>
        {model.header &&
            <h2 className="wds-global-footer-wikia-org__header">
                <a href={model.header.href} title={model.header.title.value}>
                    <Svg name={model.header['image-data'].name} className="wds-global-footer__header-logo" />
                </a>
            </h2>
        }
        <ul className="wds-global-footer-wikia-org__links">
            <li className="wds-global-footer-wikia-org__link">
                {t('global-footer-site-overview-link-wikia-inc', {year: year})}
            </li>
            {model.site_overview.links.map(link =>
                <li className="wds-global-footer-wikia-org__link">
                    <a href={link.href} data-tracking-label={link.tracking_label}>
                        {t(link.title.key)}
                    </a>
                </li>
            )}
        </ul>
    </React.Fragment>
};

export default withTranslation()(GlobalFooterWikiaOrg);
