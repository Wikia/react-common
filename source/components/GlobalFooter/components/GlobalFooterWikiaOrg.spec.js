import React from 'react';
import renderer from 'react-test-renderer';

import GlobalFooterWikiaOrg from './GlobalFooterWikiaOrg';

test('GlobalFooterWikiaOrg renders correctly', () => {
    const component = renderer.create(
        <GlobalFooterWikiaOrg model={{
            "header": {
                "type": "link-image",
                "image-data": {"type": "wds-svg", "name": "wds-company-logo-wikia-org"},
                "href": "https:\/\/www.wikia.org",
                "title": {"type": "text", "value": "Wikia.org"},
                "tracking_label": "logo"
            },
            "site_overview": {
                "links": [{
                    "type": "link-text",
                    "title": {"type": "translatable-text", "key": "global-footer-site-overview-link-terms-of-use"},
                    "href": "https:\/\/www.fandom.com\/terms-of-use",
                    "tracking_label": "site-overview.terms-of-use"
                }, {
                    "type": "link-text",
                    "title": {"type": "translatable-text", "key": "global-footer-site-overview-link-privacy-policy"},
                    "href": "https:\/\/www.fandom.com\/privacy-policy",
                    "tracking_label": "site-overview.privacy-policy"
                }, {
                    "type": "link-text",
                    "title": {"type": "translatable-text", "key": "global-footer-community-link-support"},
                    "href": "\/\/community.fandom.com\/wiki\/Special:Contact",
                    "tracking_label": "community.support"
                }, {
                    "type": "link-text",
                    "title": {"type": "translatable-text", "key": "global-footer-community-link-help"},
                    "href": "\/\/community.fandom.com\/wiki\/Help:Contents",
                    "tracking_label": "community.help"
                }]
            },
            "mobile_site_button": {
                "type": "link-text",
                "title": {"type": "translatable-text", "key": "global-footer-mobile-site-link"}
            },
            "is-wikia-org": true
        }}/>,
    );

    expect(component.toJSON()).toMatchSnapshot();
});

