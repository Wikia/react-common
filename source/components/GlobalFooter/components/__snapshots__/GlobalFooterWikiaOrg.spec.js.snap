// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFooterWikiaOrg renders correctly 1`] = `
<footer
  className="wds-global-footer-wikia-org"
>
  <h2
    className="wds-global-footer-wikia-org__header"
  >
    <a
      href="https://www.wikia.org"
      title="Wikia.org"
    >
      <div
        className="wds-global-footer__header-logo"
      />
    </a>
  </h2>
  <ul
    className="wds-global-footer-wikia-org__links"
  >
    <li
      className="wds-global-footer-wikia-org__link"
    >
      <a
        data-tracking-label="site-overview.terms-of-use"
        href="https://www.fandom.com/terms-of-use"
      >
        global-footer-site-overview-link-terms-of-use
      </a>
    </li>
    <li
      className="wds-global-footer-wikia-org__link"
    >
      <a
        data-tracking-label="site-overview.privacy-policy"
        href="https://www.fandom.com/privacy-policy"
      >
        global-footer-site-overview-link-privacy-policy
      </a>
    </li>
    <li
      className="wds-global-footer-wikia-org__link"
    >
      <a
        data-tracking-label="community.support"
        href="//community.fandom.com/wiki/Special:Contact"
      >
        global-footer-community-link-support
      </a>
    </li>
    <li
      className="wds-global-footer-wikia-org__link"
    >
      <a
        data-tracking-label="community.help"
        href="//community.fandom.com/wiki/Help:Contents"
      >
        global-footer-community-link-help
      </a>
    </li>
  </ul>
</footer>
`;
